File: C++/面试题 17.10. 主要元素.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major;
        int count = 0;

        for(auto n:nums) {
            if(count == 0) {
                major = n;
            }
            count += (major == n) ? 1 : -1;
        }

        count = 0;

        for(auto n:nums) {
            count += (major == n) ? 1 : 0;
        }

        return count > nums.size() / 2 ? major : -1;
    }
};

File: Python3/面试题 17.10. 主要元素.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # nums.sort()
        # x = nums[len(nums)//2]
        # return x if nums.count(x) > len(nums) // 2 else -1

        count = 0
        major = 0
        for n in nums:
            if count == 0:
                major = n
            count += (1 if n == major else -1)

        count = 0
        
        for n in nums:
            if(n == major):
                count += 1
        
        return (major if count > len(nums) // 2 else -1)
